com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Input/HDRP/HDSampleBufferNode.cs

```diff
@@ -0,0 +1,144 @@
+using Usage = UnityEditor.ShaderGraph.GraphDelta.GraphType.Usage;
+
+namespace UnityEditor.ShaderGraph.Defs
+{
+    internal class HDSampleBufferNode : IStandardNode
+    {
+        public static string Name => "HDSampleBuffer";
+        public static int Version => 1;
+
+        public static NodeDescriptor NodeDescriptor => new(
+            Version,
+            Name,
+            functions: new FunctionDescriptor[] {
+                new (
+                    "NormalWorldSpace",
+@"   uint2 pixelCoords = uint2(UV * _ScreenSize.xy);
+   NormalData normalData;
+   DecodeFromNormalBuffer(pixelCoords, normalData);
+   Out = normalData.normalWS;",
+                    new ParameterDescriptor[]
+                    {
+                        new ParameterDescriptor("UV", TYPE.Vec2, Usage.In, REF.ScreenPosition_Default),
+                        //I don't see the sampler param getting used anywhere.  Is it needed?
+                        new ParameterDescriptor("Sampler", TYPE.SamplerState, Usage.In),
```

Comment:
I see in the SG1 node it calls the sampler in [Line 200](https://github.com/Unity-Technologies/Graphics/blob/0c2ce3b70f80ea9549f6d091779fede2336efb52/com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Nodes/HDSampleBufferNode.cs#L200), and I tried connecting a sampler to the node and using the master stack to stay out of preview mode, it did use the sampler state in the generated shader code.
            `float3 _HDSampleBuffer_Output_2 = Unity_HDRP_SampleBuffer_NormalWorldSpace_float(float4(IN.NDCPosition.xy, 0, 0).xy, UnityBuildSamplerStateStruct(SamplerState_Linear_Repeat).samplerstate);`